components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java

```diff
@@ -1304,4 +1323,94 @@ protected void createLegacyUser(OAuthTokenReqMessageContext tokReqMsgCtx, Assert
         tokReqMsgCtx.setAuthorizedUser(user);
     }
 
+    /**
+     * Validate the signature against the certificate obtained from SAML Sign KeyStore which is defined under
+     * Security.SAMLSignKeyStore in carbon.xml.
+     *
+     * @param assertion
+     * @throws IdentityOAuth2Exception
+     */
+    protected void validateSignatureAgainstSAMLSignKeyStoreCertificate(Assertion assertion)
+            throws IdentityOAuth2Exception {
+
+        try {
+            X509Certificate x509Certificate = getCertificateFromSAMLSignKeyStore();
+            X509Credential x509Credential = new X509CredentialImpl(x509Certificate);
+            SignatureValidator signatureValidator = new SignatureValidator(x509Credential);
+            signatureValidator.validate(assertion.getSignature());
+        } catch (ValidationException e) {
+            if (StringUtils.isNotEmpty(assertion.getIssuer().getValue())) {
+                throw new IdentityOAuth2Exception(
+                        "Error while validating the signature from SAML sign keystore for SAML Token Issuer :  "
+                                + assertion.getIssuer().getValue(), e);
+            } else {
+                throw new IdentityOAuth2Exception(
+                        "Error while validating the signature from SAML sign keystore. The SAML Token Issuer is null",
+                        e);
+            }
+        }
+    }
+
+    /**
+     * Get the certificate from the SAML Sign KeyStore which is defined under Security.SAMLSignKeyStore in carbon.xml
+     *
+     * @throws IdentityException
+     */
+    private X509Certificate getCertificateFromSAMLSignKeyStore() throws IdentityOAuth2Exception {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Getting the certificate from separate SAMLSignKeyStore");
```

Comment:
Add full-stop at the end.

Reply:
Addressed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1182